webapp/components/user_settings/email_notification_setting.jsx

```diff
@@ -37,9 +37,16 @@ export default class EmailNotificationSetting extends React.Component {
                 emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_FIFTEEN_MINUTES)
             };
         } else {
-            // otherwise, the default interval is immediately
+            /**
+             * otherwise, the default interval is immediately
+             *
+             * `emailInterval` in the backend is irrespective of whether email Bacthing is enabled or not.
+             * Hence this check is kept in the component to choose the default interval as `immediately` when
+             * `EnableEmailBatching` is false.
+             */
+
             this.state = {
-                emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_IMMEDIATE)
+                emailInterval: Preferences.INTERVAL_IMMEDIATE
```

Comment:
While that's true, you're introducing strong coupling by assuming that the component will do the right thing, and by assuming that there will only ever be two options here: immediate, and never.

Instead, would it be possible to fetch the preference, and then check if it makes sense (i.e. is it set to never or immediate), and if it doesn't make sense, then coerce it to immediate instead? This approach would add a bit of complexity, but I think it might be more clear to somebody reading this code for the first time. You could do the same thing for when batching is enabled - check if the setting is one of immediate, 15 minutes, 1 hour, or never) and if it isn't, coerce it to 15 minutes.